src/ansys/hps/client/client.py

```diff
@@ -165,6 +171,30 @@ def __init__(
                 requests.packages.urllib3.exceptions.InsecureRequestWarning
             )
 
+        self.auth_url = auth_url
+
+        if not auth_url:
+            with requests.session() as session:
+                session.verify = self.verify
+                jms_info_url = url.rstrip("/") + "/jms/api/v1"
+                resp = session.get(jms_info_url)
+                if resp.status_code != 200:
+                    raise RuntimeError(
+                        f"Failed to contact jms info endpoint {jms_info_url}, \
+                            status code {resp.status_code}: {resp.content.decode()}"
+                    )
+                else:
+                    jms_data = resp.json()
+                    if "services" in jms_data and "external_auth_url" in jms_data["services"]:
+                        self.auth_url = resp.json()["services"]["external_auth_url"]
+                    else:
+                        log.warning(
```

Comment:
I will move the warning message to "if not realm".  That way its just more obvious if anyone messes up their configuration.  I presume they should not be setting the realm manually unless they know what they are doing, but applications like the scaler expose it and it might be improperly configured by the customer to point to an old instance but not have realm specified.